rdt/transformers/categorical.py

```diff
@@ -182,21 +172,23 @@ def _transform(self, data):
         Returns:
             numpy.ndarray
         """
+        fit_categories = pd.Series(self.intervals.keys())
+        has_nan = pd.isna(fit_categories).any()
+        unseen_indexes = ~(data.isin(fit_categories) | (pd.isna(data) & has_nan))
+        if unseen_indexes.any():
+            unseen_categories = set(data[unseen_indexes])
+            warnings.warn(
+                f'Warning: The data contains new categories {unseen_categories} that were not '
```

Comment:
Should we limit the `unseen_categories` message to the first `5` since otherwise we will flood the log... ?
Also, do we want to keep the `Warning:` ? The message output ends up being like this:
```
UserWarning: Warning: The data contains new categories {'hello', 'world'} that were not seen in the original data. Assigning them random values. If you want to model new categories, please fit the transformer again with the new data.
```